benchmarks/tensors/dibaryon/baryon_generator.cpp

```diff
@@ -47,168 +210,329 @@ void generate_function(std::string name)
+
+      complex_expr bdouble_update_def =
+        Bdouble_init(t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2) +
+        o * prop(t, 0, iCprime, iSprime, ic, is, x2, y) * psi;
+      complex_computation bdouble_update(
+          // name
+          str_fmt("bdouble_o_update_%d_%d", ic, is),
+          // iterator
+          {t, jCprime, jSprime, kCprime, kSprime, x, n, iCprime, iSprime, x2, y},
+          // definition
+          bdouble_update_def);
+
+      Bdouble_o_updates.push_back(bdouble_update);
+
+      computation *o_real;
+      computation *o_imag;
+      std::tie(o_real, o_imag) = iAndComp.second;
+      O2UserEdge edge {o_real, o_imag, bdouble_update.get_real(), bdouble_update.get_imag()};
+      o2userEdges.push_back(edge);
+    }
 
-    computation Bdouble_i_update1("Bdouble_i_update1", {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, lCprime, lSprime, x, x2, y},
-	    Bdouble_i_init(t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2) - mul_i(P_update, prop_2pp));
+#define Q_BLOCK_SIZE 3
+#define O_BLOCK_SIZE 3
+#define P_BLOCK_SIZE 3
 
     // -------------------------------------------------------
     // Layer II
     // -------------------------------------------------------
-    block init_blk({&Blocal_r_init, &Blocal_i_init, &Q_r_init, &Q_i_init,
-		&Bsingle_r_init, &Bsingle_i_init, &Bdouble_r_init, &Bdouble_i_init,
-		&O_r_init, &O_i_init, &P_r_init, &P_i_init});
-
-    block Blocal_blk({&Blocal_r_update, &Blocal_i_update, &Q_r_update, &Q_i_update,
-		 &O_r_update, &O_i_update, &P_r_update, &P_i_update});
-
-    block Bsingle_blk({&Bsingle_r_update, &Bsingle_i_update, &Bdouble_r_update0,
-			&Bdouble_i_update0, &Bdouble_r_update1, &Bdouble_i_update1});
-
-#if FUSE
-    Bsingle_blk.interchange(iCprime, jCprime);
-    Bsingle_blk.interchange(iSprime, jSprime);
-    Bsingle_blk.interchange(iCprime, kCprime);
-    Bsingle_blk.interchange(iSprime, kSprime);
-#endif
 
-    Blocal_r_init.then(Blocal_i_init, x)
-		 .then(Q_r_init, computation::root)
-		 .then(Q_i_init, y)
-		 .then(Bsingle_r_init, x2)
-		 .then(Bsingle_i_init, x2)
-		 .then(Bdouble_r_init, x2)
-		 .then(Bdouble_i_init, x2)
-		 .then(O_r_init, y)
-		 .then(O_i_init, y)
-		 .then(P_r_init, y)
-		 .then(P_i_init, y)
-		 .then(Blocal_r_update, computation::root)
-		 .then(Blocal_i_update, wnum)
-		 .then(Q_r_update, jSprime)
-		 .then(Q_i_update, wnum)
-		 .then(O_r_update, wnum)
-		 .then(O_i_update, wnum)
-		 .then(P_r_update, wnum)
-		 .then(P_i_update, wnum)
-		 .then(Bsingle_r_update, n)
-		 .then(Bsingle_i_update, y)
-		 .then(Bdouble_r_update0, y)
-		 .then(Bdouble_i_update0, y)
-		 .then(Bdouble_r_update1, y)
-		 .then(Bdouble_i_update1, y);
-
-
-    if (PARALLEL)
-    {
-        Blocal_r_init.tag_parallel_level(t);
-        Blocal_r_update.tag_parallel_level(t);
-        Bsingle_r_update.tag_parallel_level(t);
+    computation *handle = &(
+        Blocal_r_init
+        .then(Blocal_i_init, computation::root)
```

Comment:
I'm not sure this is the best order (root level ordering). Did you try to fuse them? If you follow the same old order you might get better performance. I tried many combinations of fusion and that order gave me the best performance on Lanka.